segment.go

```diff
@@ -101,6 +101,11 @@ type PostingsIterator interface {
 	Size() int
 }
 
+type BytesOffDiskStats interface {
```

Comment:
Better naming for this please, perhaps -

```
type DiskStats interface {
    NumReadBytes() uint64
    UpdateReadBytes(uint64)
}
```

or any other better names.

Reply:
How about the following?
```
type DiskStatsReporter interface {
	NumBytesRead() uint64
	SetBytesRead(uint64)
}
```

Reply:
NumBytesRead()  Vs BytesRead() ?  Isn't the latter is enough?

Reply:
Yeah I think that should suffice.